source/loader/layers/sanitizer/asan_libdevice.hpp

```diff
@@ -82,6 +82,10 @@ struct LaunchInfo {
 constexpr unsigned ASAN_SHADOW_SCALE = 3;
 constexpr unsigned ASAN_SHADOW_GRANULARITY = 1ULL << ASAN_SHADOW_SCALE;
 
+// Based on the observation, only the last 24 bits of the address of the private
+// variable have changed, we use 31 bits(2G) to be safe.
+constexpr std::size_t ASAN_PRIVATE_SIZE = 0x7fffffffULL + 1;
```

Comment:
2G is probably too big, how about just use 24 bits?

Reply:
We should set ASAN_PRIVATE_SIZE as large as possible, if not, it will produce unpredictable wrong error report.

Reply:
is there a reproducer that produces a wrong report if we use 24 bits?

Reply:
I tried a create a kernel that has large private memory but failed at "PI_OUT_OF_RESOURCES" when the private memory size is more than 8MB without asan.
Besides, if asan is enabled, the maximum size of private memory on user side should be small.

Therefore, I think it's fine to use 24 bits now.

Reply:
Thank you for the test. Using 24bit lgtm. Though it will still cause allocating a huge memory, which may result in OOM, when number of workgroups is large. I guess we probably need to reduce ASAN_PRIVATE_SIZE further in the future.

Reply:
I think we can also prevent from OOM by limiting the detecting range of workgroup